quickstart/src/main/java/com/twilio/video/quickstart/util/CameraCapturerCompat.java

```diff
@@ -151,9 +146,24 @@ private boolean isPrivateImageFormatSupportedForCameraId(String cameraId) {
             e.printStackTrace();
             return false;
         }
+        /*
+         * This is a temporary work around for a RuntimeException that occurs on devices which contain cameras
+         * that do not support ImageFormat.PRIVATE output formats. A long term fix is currently in development.
+         * https://github.com/twilio/video-quickstart-android/issues/431
+         */
         final StreamConfigurationMap streamMap =
                 cameraCharacteristics.get(CameraCharacteristics.SCALER_STREAM_CONFIGURATION_MAP);
         isPrivateImageFormatSupported = streamMap.isOutputSupportedFor(ImageFormat.PRIVATE);
-        return isPrivateImageFormatSupported;
+
+        /*
+         * Read the color filter arrangements of the camera to filter out the ones that support
+         * SENSOR_INFO_COLOR_FILTER_ARRANGEMENT_MONO or SENSOR_INFO_COLOR_FILTER_ARRANGEMENT_NIR.
+         * Visit this link for details on supported values - https://developer.android.com/reference/android/hardware/camera2/CameraCharacteristics#SENSOR_INFO_COLOR_FILTER_ARRANGEMENT
+         */
+        final int colorFilterArrangement = cameraCharacteristics.get(CameraCharacteristics.SENSOR_INFO_COLOR_FILTER_ARRANGEMENT);
+
+        isMonoChromeSupported = (colorFilterArrangement == 5 || colorFilterArrangement == 6) ? true : false;
```

Comment:
Instead of 5 and 6 can you reference the constants defined in the Android framework